component/src/main/java/org/wso2/extension/siddhi/execution/extrema/KalmanMinMaxStreamProcessor.java

```diff
@@ -0,0 +1,245 @@
+                    StreamEvent minimumEvent;
+
+                    switch (extremaType) {
+                        case MINMAX:
+                            maximumEvent = getMaxEvent(output);
+                            minimumEvent = getMinEvent(output);
+                            if (maximumEvent != null && minimumEvent != null) {
+                                if (maxEventPos > minEventPos) {
+                                    returnEventChunk.add(minimumEvent);
+                                    returnEventChunk.add(maximumEvent);
+                                } else {
+                                    returnEventChunk.add(maximumEvent);
+                                    returnEventChunk.add(minimumEvent);
+                                }
+                            } else if (maximumEvent != null) {
+                                returnEventChunk.add(maximumEvent);
+                            } else if (minimumEvent != null) {
+                                returnEventChunk.add(minimumEvent);
+                            }
+                            break;
+                        case MIN:
+                            minimumEvent = getMinEvent(output);
+                            if (minimumEvent != null) {
+                                returnEventChunk.add(minimumEvent);
+                            }
+                            break;
+                        case MAX:
+                            maximumEvent = getMaxEvent(output);
+                            if (maximumEvent != null) {
+                                returnEventChunk.add(maximumEvent);
+                            }
+                            break;
+                    }
+                    eventStack.remove();
+                    valueStack.remove();
+                }
+            }
+        }
+        if (returnEventChunk.getFirst() != null) {
+            nextProcessor.process(returnEventChunk);
+        }
+    }
+
+    private StreamEvent getMinEvent(Queue<Double> output) {
+        // value 2 is an optimized value for stock market domain, this value may change for other domains
+        Integer smoothenedMinEventPosition = extremaCalculator.findMin(output, 2);
+        if (smoothenedMinEventPosition != null) {
+            //value 10 is an optimized value for stock market domain, this value may change for other domains
+            Integer minEventPosition = extremaCalculator.findMin(valueStack, 10);
+            if (minEventPosition != null) {
+                StreamEvent returnMinimumEvent = getExtremaEvent(minEventPosition);
+                if (returnMinimumEvent != null) {
+                    minEventPos = minEventPosition;
+                    complexEventPopulater.populateComplexEvent(returnMinimumEvent, new Object[]{"min"});
+                    return returnMinimumEvent;
+                }
+            }
+        }
+        return null;
+    }
+
+    private StreamEvent getMaxEvent(Queue<Double> output) {
```

Comment:
Shall we add method level comments to get a better understanding of each method?